guides/traversing-pages.rst

```diff
@@ -1,164 +1,200 @@
 
-XPath Selectors
-~~~~~~~~~~~~~~~
+Nested Traversing
+~~~~~~~~~~~~~~~~~
 
-And of course, you can use clean XPath queries:
+Every ``find*()`` method will return a ``Behat\Mink\Element\NodeElement`` instance
+and ``findAll()`` will return an array of such instances. The fun part is
+you can make same old traversing on such elements too:
 
 .. code-block:: php
 
-    $xpath = $handler->selectorToXpath('xpath', '//html');
+    $registerForm = $page->find('css', 'form.register');
 
-It's like a proxy method, which will return the same expression you give
-to it. It's used internally in `find* methods`_.
+    if (null === $registerForm) {
+        throw new \Exception('The element is not found');
+    }
 
-``find*`` Methods
-~~~~~~~~~~~~~~~~~
+    // find some field INSIDE form with class="register"
+    $field = $registerForm->findField('Email');
+
+.. _selectors:
+
+Selectors
+---------
 
-So, now we know how to generate XPath queries for specific elements search.
-But how we actually make this search? The answer is ``find*`` methods,
-available on ``DocumentElement`` object. You can get this object from session:
+The ``ElementInterface::find`` and ``ElementInterface::findAll`` methods
+support several kinds of selectors to find elements.
+
+CSS Selector
+~~~~~~~~~~~~
+
+The ``css`` selector type lets you use CSS expressions to search elements
+in the page:
 
 .. code-block:: php
 
-    $page = $session->getPage();
-    $page = $mink->getSession('sahi')->getPage();
+    $title = $page->find('css', 'h1');
 
-This object provides two very useful traversing methods:
+XPath Selector
+~~~~~~~~~~~~~~
 
-* ``find()`` - evaluates specific selector on the page content and returns
-  the last matched element or ``null``:
+The ``xpath`` selector type lets you use XPath queries to search elements
+in the page:
 
-  .. code-block:: php
+.. code-block:: php
 
-      $fieldElement = $page->find('named',
-          array('field', 'id|name|value|label')
-      );
-      $elementByCss = $page->find('css', 'h3 > a');
+    $anchorsWithoutUrl = $page->findAll('xpath', '//a[not(@href)]');
 
-* ``findAll()`` - evaluates specific selector on the page content and returns
-  an array of matched elements:
+.. caution::
 
-  .. code-block:: php
+    This selector searches an element inside the current node (which is ``<html>``
+    for the page object). This means that trying to pass it the XPath of
+    and element retrieved with ``ElementInterface::getXpath`` will not work
+    (this query includes the query for the root node). To check whether an
+    element object still exists in the browser page, use ``ElementInterface::isValid``
+    instead.
 
-      $fieldElements = $page->findAll('named',
-          array('field', 'id|name|value|label')
-      );
-      $elementsByCss = $page->findAll('css', 'h3 > a');
+.. _named-selector:
 
-Also, there's a bunch of shortcut methods:
+Named Selector
```

Comment:
Maybe use plural form here as well because you start next sentence with `Named selectors`.